arangod/Network/Methods.cpp

```diff
@@ -234,6 +234,13 @@ class RequestsState final : public std::enable_shared_from_this<RequestsState> {
 
   // scheduler requests that are due
   void startRequest() {
+    if (!_pool) {
+      LOG_TOPIC("5949f", ERR, Logger::COMMUNICATION)
+          << "connection pool unavailable";
+      callResponse(Error::Canceled, nullptr, std::move(_request));
+      return;
+    }
+
     auto now = std::chrono::steady_clock::now();
     if (now > _endTime || _pool->config().clusterInfo->server().isStopping()) {
```

Comment:
very good, fixed this hidden nullptr issue.
Although I do not think it actually happens.